src/Features/Auth/ducks.js

```diff
@@ -63,7 +65,16 @@ export const authReducer = (state = { ...initialAuthState }, action) => {
         contact,
       };
     }
+    case UPDATE_ROLE: {
+      const { role } = action.payload;
 
+      const { role: previousrole } = state;
+      let isUserRoleChange = false;
+      if (previousrole !== role) {
+        isUserRoleChange = true;
+      }
+      return { ...state, role, isUserRoleChange };
```

Comment:
{...state, role, isUserRoleChange: previousRole !== role } this will do what you want to do but with precise code.